### PR TITLE
build: add bazel config for cdk-experimental/dialog

### PR DESCRIPTION
**Pull request by crisbeto**
Adds the BUILD.bazel file for the `@angular/cdk-experimental/dialog` package for consistency with everything else and to make it easier to eventually move it into the CDK.

### CODE REVIEW

File: src/cdk-experimental/dialog/typings.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+declare var module: {id: string};
```
Comment (crisbeto) [inline src/cdk-experimental/dialog/typings.d.ts]: A note on this one: I still can't find a way to get Bazel to pick up the `typings.d.ts` that is one level up in `cdk-experimental`. This is also something that came up in https://github.com/angular/material2/pull/11481.

Comment (jelbourn) [reply src/cdk-experimental/dialog/typings.d.ts]: That's expected; Bazel doesn't allow referencing source files that aren't in the package's subtree; if you wanted to reference a common `d.ts` file, you have to create a build rule for it and depend on that.

**Comment by josephperrott** [comment]
@crisbeto Please rebase when you can

**Comment by crisbeto** [comment]
Rebased.

**Pull request merged**